providers/sgx/sgx-app/src/sgx_app/cloud.rs

```diff
@@ -55,18 +56,12 @@ pub fn generate_keypair(
     let mut report_data = [0u8; 64];
     report_data[0..32].copy_from_slice(&result);
     let report = Report::for_target(&targetinfo, &report_data);
-    let pkcs1 = priv_key.to_pkcs1();
+    let pkcs1 = priv_key.to_pkcs1_der();
     priv_key.zeroize();
-    if let Ok(mut secret) = pkcs1 {
-        match crate::sgx_app::keypair_seal::seal_secret(csprng, &secret, result.into()) {
-            Ok(sealed_secret) => {
-                secret.zeroize();
-                Ok((pub_key, sealed_secret, report))
-            }
-            Err(e) => {
-                secret.zeroize();
-                Err(CloudError::SealingError(e))
-            }
+    if let Ok(secret) = pkcs1 {
+        match crate::sgx_app::keypair_seal::seal_secret(csprng, secret.as_der(), result.into()) {
+            Ok(sealed_secret) => Ok((pub_key, sealed_secret, report)),
+            Err(e) => Err(CloudError::SealingError(e)),
```

Comment:
does the `secret` have zeroize on drop?

Reply:
yes, it's a wrapper of `Zeroizing<Vec<u8>>`